### ISSUE TITLE
PAGE FAULT IN NON PAGED AREA with nvlddmkm.sys when trying to use godot to view models in VR.

### ISSUE THREAD

**Issue by RileyGuy** [godotengine/godot#23319]
**Godot version:**
3.0.6

**OS/device including version:**
Windows 10 LTSB Version 1607 (OS Build 14393.2580)
GTX 970 with driver version 416.34 from Nvidia.

**Issue description:**
I was using Godot to view some models in VR to get a sense of their scale, when randomly, be it when the project runs, SteamVR runs, or when something runs. (It even happened when I touched my rift touch controllers one time and pushed a button or two.) My computer blue screens with the aforementioned error in the title. I've contacted Oculus and narrowed it down to godot itself. It only happens after I run the project. There is nothing funky about the project either, it's all just models and textures.

**Steps to reproduce:**
Make a minimal VR environment within Godot, maybe import some models and textures with them, etc., run the project, close it, and the crash may happen when one of a few things occur: When you re-open the project. When you start steamVR. When you touch your oculus controllers. When you touch the oculus home button. There are probably more instances where this can happen, but that is all I've encountered. My computer has blue-screened probably 10 times within the past day or two. I'm unsure if this will be reproduce-able on other hardware, as my friend has no problems with his Vive.

In conclusion: PLEASE help me fix this, I just want to view models in VR without my computer crashing at random intervals.

**Comment by akien-mga** [comment]
CC @Mux213

**Comment by OvermindDL1** [comment]
> My computer blue screens with the aforementioned error in the title. 

This should be entirely outright impossible for any usermode application, so I don't see how this could potentially be Godot's issue at all.  A kernel panic (blue screen) means that something in unprotected ring 0 failed, which should just categorically not happen or it is a major system bug, not a userland bug.  Likely this is caused by a bad driver or failing hardware, and based on the error of `nvlddmkm.sys` (nvidia ddm kernel module) it's either nvidia's driver has a bug (rather common the last few years) or the GPU itself is failing.

**Comment by OvermindDL1** [comment]
Thus the first question, have you tried newer or older nvidia drivers?

**Comment by RileyGuy** [comment]
It seems to have been an issue with my drivers, yeah. I downgraded from the later 4xx drivers to 391.35 and that seems to have fixed it.

**Comment by akien-mga** [comment]
Thanks, closing then.

**Comment by RileyGuy** [comment]
The issue is actually back, if you could please re-open this. I'm getting the same error even with downgraded drivers. I have absolutely no clue as to what's happening.

**Comment by akien-mga** [comment]
Sure, but it's still unlikely to be a Godot bug. If drivers are not at fault, then hardware might be.

**Comment by OvermindDL1** [comment]
If it is really so random I'd be leaning more toward a heat issue in the hardware.  If it's not random then it's likely still a driver fault.  Not something Godot could fix in either case...  :-/

**Comment by RileyGuy** [comment]
> 
> 
> If it is really so random I'd be leaning more toward a heat issue in the hardware. If it's not random then it's likely still a driver fault. Not something Godot could fix in either case... :-/

I have no heat issues, and all other programs work fine. It only happens with Godot running. No other programs are influencing my VR experience. It's simply SteamVR, Oculus, and Godot running.

**Comment by OvermindDL1** [comment]
Regardless though, there is no way a user-mode application could cause a kernel panic unless there is a bug in something on Ring-0, thus being the kernel, drivers, or hardware.

**Comment by RileyGuy** [comment]
Well, it seems to *only* happen with godot present, no other programs cause any problems. Is there something I can do to narrow it down to what's actually happening underneath?

**Comment by OvermindDL1** [comment]
> Well, it seems to only happen with godot present, no other programs cause any problems.

That would just be because Godot is calling some rendering path that whatever other programs you are testing are not, regardless of how it is called it should never kernel panic, that is a kernel/driver/hardware bug.

As for testing it, if you are on Windows then 'good luck' is generally the only method (kernel debugging is hell upon itself, it's *much* easier on linux).  As for a workaround just keep trying different driver versions until something works...  :/

To hardware test then you might try a linux live usb stick with the official nvidia binary blob drivers and try Godot on that?  It may or may not test the same paths depending on how nvidia implements their binary blobs (they aren't open like AMD's and Intel's) but it's the best try currently?

**Comment by RileyGuy** [comment]
I could try that, but I'd have no way of testing it with VR or not, since oculus is not supported on linux, and I don't have a vive. But I could try running a normal project on Windows to see if the bluescreen still happens. Though it seems to happen whenever I interact with one of the two VR platforms after the project has ran, even after the project is done running. I've determined it's probably not oculus, it seems like SteamVR might be handling things badly in regards to either godot, or handing godot's stuff over to oculus.

**Comment by RileyGuy** [comment]
Actually, something interesting just happened. I had Godot and the Unreal editor open, and without even touching the VR button on Godot, when I tried previewing my project in UE, it bluescreened again. So it seems like even just having Godot open causes it to crash when *anything* touches VR.

**Comment by BastiaanOlij** [comment]
One thing to look at is the gdnlib file and see if reloadable is set to false.
The VR plugins do not like it when they are unloaded when still active because the focus moves away from Godot.

**Comment by BastiaanOlij** [comment]
Is this still an issue? I haven't heard anyone experiencing this anymore after we disabled the reloadable property

**Comment by akien-mga** [comment]
Assuming fixed in 3.1, please comment if you can still reproduce this issue.
